### ISSUE TITLE
Remove deprecated twisted.cred.util

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4107]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4107|
|Type|task|
|Created|2009-11-13 20:51:09Z|
|Branch|https://github.com/twisted/twisted/tree/remove-cred-util-4107|

The docstring of [twisted/cred/util.py](twisted/cred/util.py) mentions:

```
Outdated, deprecated functionality related to challenge-based authentication.

Seek a solution to your problem elsewhere.  This module is deprecated.
```

Let's get rid of it asap.

Attachments:

* [4107.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d56/d56fe91441a94395b6e0c3143b75aa76c375c79e/2a5068d1411f96f408f41a2565e1df08d2821b08.diff) (4232 bytes) - added by cyli on 2010-05-26 00:17:08Z - Removed twisted.cred.util and its corresponding deprecation test in tests/test_newcred.py

<details><summary>Searchable metadata</summary>

```
trac-id__4107 4107
type__task task
reporter__thijs thijs
priority__highest highest
milestone__ 
branch__branches_remove_cred_util_4107 branches/remove-cred-util-4107
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1258145469000000 1258145469000000
changetime__1284665707000000 1284665707000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @thijstriemstra|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)|@cyli removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_4"></a>|@mithrandi set owner to @cyli|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli set status to `closed`|
|-|-|

(In [29095]) Apply patch 4107.diff from ticket 4107: remove twisted.cred.util

Author: cyli
Reviewer: mithrandi
Fixes: [#4107](https://github.com/twisted/twisted/issues/4107)

Removed twisted.cred.util and its corresponding deprecation test in tests/test_newcred.py.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `reopened`|
|-|-|

Reopened in r30023

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [30023]) Revert r29095 - incorrectly applied patch

Reopens [#4107](https://github.com/twisted/twisted/issues/4107)

The file to be removed, `twisted/cred/util.py`, was not actually removed.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [30024]) Branching to 'remove-cred-util-4107'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [30026]) Merge remove-cred-util-4107

Author: cyli
Reviewer: mithrandi
Fixes: [#4107](https://github.com/twisted/twisted/issues/4107)

Remove `twisted.cred.util` and its corresponding deprecation tests.  This
re-merge corrects the fact that previously only the contents of
`twisted/cred/util.py` were removed, not the file itself.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
